File: BE-SpringBoot/src/main/java/com/example/mugbackend/analysis/repository/AnalysisRepository.java
package com.example.mugbackend.analysis.repository;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import com.example.mugbackend.analysis.domain.Analysis;

@Repository
public interface AnalysisRepository extends MongoRepository<Analysis, String>, CustomAnalysisRepository {

}

File: app/src/main/java/com/example/mulga/data/repository/AnalysisRepository.kt
package com.example.mulga.data.repository

import com.example.mulga.data.service.AnalysisService
import com.example.mulga.domain.mapper.toDomainModel
import com.example.mulga.domain.model.AnalysisEntity

class AnalysisRepository(private val analysisService: AnalysisService) {

//    suspend fun getAnalysisData(year: Int, month: Int): AnalysisEntity? {
//        val response = analysisService.getAnalysis(year, month)
//        return if (response.isSuccessful) response.body()?.toDomainModel() else null
//    }
}
